mash_client/cli/job/azure.py

```diff
@@ -40,12 +41,17 @@ def azure():
 
 
 @click.command()
+@click.option(
+    '--dry-run',
```

Comment:
Why not `--validate`, does any other checking happen than job-doc validation?

Reply:
I think `--validate` could be confusing from user perspective. "Is validation normally off and the flag turns it on?" Something like `--validate-only` would be more expressive to what the flag is used for.

`--dry-run` seems as good as `--validate-only` though. My preference would be leave as is mainly to prevent the effort of renaming it in this PR and in server PR given both options seem okay.